### PR TITLE
DAOS-12425 control: Add opt to auto-format on server start-up

### PR DESCRIPTION
**Pull request by tanabarr**
Add daos_server start --auto option to enable engines to start without
having to trigger a storage format with an externally executed dmg
storage format command. If the flag is detected the server will call
the format API during its start-up routine and if the server config
file used is valid the engines will be started after format completes.

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by kjacque — APPROVED** [review]
Nice and simple. Did you test it manually?

IMO we need system tests for this scenario, but I am okay with adding them later.

File: src/control/cmd/daos_server/start.go [hunk 1]
```diff
@@ -27,13 +27,14 @@ type startCmd struct {
...
+	Targets             uint16  `short:"t" long:"targets" description:"Number of targets to use (default use all cores)"`
+	NrXsHelpers         *uint16 `short:"x" long:"xshelpernr" description:"Number of helper XS per VOS target"`
+	FirstCore           uint16  `short:"f" long:"firstcore" default:"0" description:"Index of first core for service thread"`
 	Group               string  `short:"g" long:"group" description:"Server group name"`
 	SocketDir           string  `short:"d" long:"socket_dir" description:"Location for all daos_server & daos_engine sockets"`
-	Insecure            bool    `short:"i" long:"insecure" description:"allow for insecure connections"`
-	RecreateSuperblocks bool    `long:"recreate-superblocks" description:"recreate missing superblocks rather than failing"`
+	Insecure            bool    `short:"i" long:"insecure" description:"Allow for insecure connections"`
+	RecreateSuperblocks bool    `long:"recreate-superblocks" description:"Recreate missing superblocks rather than failing"`
+	Auto                bool    `long:"auto" description:"Automatically generate server config used to start service"`
```
Comment (knard-intel) [inline src/control/cmd/daos_server/start.go:37]: Not sure to understand this comment. This Auto setting is for automatically formating at startup and not generating server config ?
If yes, this comment should probably be changed and the name of the field could be change to AutoFormat

Comment (tanabarr) [reply src/control/cmd/daos_server/start.go:37]: yes I think you're right and this should be renamed and the comment cleaned up

Comment (tanabarr) [reply src/control/cmd/daos_server/start.go]: done

**Comment by tanabarr** [comment]
> Nice and simple. Did you test it manually?
> 
> IMO we need system tests for this scenario, but I am okay with adding them later.

I manually tested a previous iteration of the PR and it works fine, it just goes ahead and formats without waiting for dmg intervention and engines get started up as long as there are no problems with config validation or the format itself. I will give it another run through manually when our clusters are back up and running. The usecase @mjmac  is interested in is with the external tmpfs so hopefully you can also give it a go @mjmac . In that case the SCM should be a no-op but the NVMe devices should still be wiped during the implicit format phase.

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by kjacque — APPROVED** [review]

**Comment by mjmac** [comment]
In my light testing here, this seems to work as expected with an existing empty tmpfs. The `daos_nvme.conf` file is created and the engines are started.

**Pull request merged**